src/static/helpers/slasHelper.test.ts

```diff
@@ -0,0 +1,188 @@
+  const expectedChallenge = 'AH8WaHxbEtoZuFw-rw2YS9SazhKJilGoESoSlICXsQw';
+
+  test('generates correct code challenge for verifier', async () => {
+    const challenge = await slasHelper.generateCodeChallenge(verifier);
+    expect(challenge).toBe(expectedChallenge);
+  });
+});
+
+describe('Get code and usid', () => {
+  const url =
+    'https://localhost:3000/callback?usid=048adcfb-aa93-4978-be9e-09cb569fdcb9&code=J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o';
+
+  const expectedRecord = {
+    code: 'J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o',
+    url,
+    usid: '048adcfb-aa93-4978-be9e-09cb569fdcb9',
+  };
+
+  test('extracts code and usid from url', () => {
+    const record = slasHelper.getCodeAndUsidFromUrl(url);
+    expect(record).toStrictEqual(expectedRecord);
+  });
+});
+
+describe('Authorize user', () => {
+  const expectedCode = 'J2lHm0cgXmnXpwDhjhLoyLJBoUAlBfxDY-AhjqGMC-o';
+
+  test('hits the authorize endpoint and recieves authorization code', async () => {
+    const authCode = (
+      await slasHelper.authorize(mockSlasClient, mockParameters)
+    ).code;
+    expect(authorizeCustomerMock).toHaveBeenCalled();
+    expect(authCode).toBe(expectedCode);
+  });
+});
+
+describe('Guest user flow', () => {
+  test('uses code challenge to generate auth code', async () => {
+    await slasHelper.loginGuestUser(mockSlasClient, mockParameters);
+
+    // Pulling out the arguments to test
+    const parametersArg = (
+      authorizeCustomerMock.mock.calls[0][0] as Parameters<
+        ShopperLogin<any>['authorizeCustomer'] // TODO: replace any with proper type
+      >[0]
+    )?.parameters;
+
+    // required query parameters
+    // expect(parametersArg?.hint).toBe('guest'); TODO: revise this line
+    expect(parametersArg?.response_type).toBe('code');
+    expect(parametersArg).toHaveProperty('client_id');
+    expect(parametersArg).toHaveProperty('code_challenge');
+    expect(parametersArg).toHaveProperty('redirect_uri');
+
+    expect(authorizeCustomerMock).toBeCalled();
+  });
+
+  test('uses auth code and code verifier to generate JWT', async () => {
+    const accessToken = await slasHelper.loginGuestUser(
+      mockSlasClient,
+      mockParameters
+    );
+
+    // Pulling out the arguments to test
+    const tokenBodyArg = (
+      getAccessTokenMock.mock.calls[0][0] as Parameters<
+        ShopperLogin<any>['getAccessToken']
+      >[0]
+    ).body;
+
+    // No authorization header is required
+    // required query parameters
+    expect(tokenBodyArg.grant_type).toBe('authorization_code_pkce');
+    expect(tokenBodyArg).toHaveProperty('code_verifier');
+    expect(tokenBodyArg).toHaveProperty('code');
+    expect(tokenBodyArg).toHaveProperty('client_id');
+    expect(tokenBodyArg).toHaveProperty('redirect_uri');
+
+    expect(accessToken).toStrictEqual({access_token: 'access_token'});
+  });
+});
+describe('Registered B2C user flow', () => {
```

Comment:
Same comment w.r.t what to test / `expect.objectContaining`. Also consider [`expect.stringMatching`](https://jestjs.io/docs/expect#expectstringmatchingstring--regexp)!